src/Yarhl.UnitTests/FileSystem/NodeContainerFormatTests.cs

```diff
@@ -183,6 +183,31 @@ public void MoveChildrenToNodeCanMergeContainers()
             Assert.That(folder3.Children.Count, Is.Zero);
         }
 
+        [Test]
+        public void Clone()
+        {
+            using NodeContainerFormat format = new NodeContainerFormat();
+
+            using Node child1 = NodeFactory.CreateContainer("child1");
+            using Node child2 = new Node("child2");
+            using Node grandchild = new Node("grandchild");
+
+            format.Root.Add(child1);
+            format.Root.Add(child2);
+            child1.Add(grandchild);
+
+            using NodeContainerFormat clone = (NodeContainerFormat)format.DeepClone();
+
+            Node child1Clone = clone.Root.Children["child1"];
+            Node child2Clone = clone.Root.Children["child2"];
+            Node grandchildClone = child1Clone.Children["grandchild"];
+
+            Assert.AreNotSame(format, clone);
+            Assert.AreNotSame(child1, child1Clone);
+            Assert.AreNotSame(child2, child2Clone);
+            Assert.AreNotSame(grandchild, grandchildClone);
```

Comment:
Maybe this is passing because `grandchildClone` is `null`? I think the copy is not recursively.

Reply:
No, `grandchildClone` has contents:

![image](https://user-images.githubusercontent.com/5345429/117547241-ef050580-b02e-11eb-9f0e-178ab88321ad.png)